express-notes-api/server.js

```diff
@@ -0,0 +1,121 @@
+  const notes = data.notes;
+  const id = Number(req.params.id);
+  if (id < 1 || !Number.isInteger(id) || isNaN(id)) {
+    res.status(400).send({ error: 'id must be a positive integer' });
+  } else if (!notes[id]) {
+    res.status(404).send({ error: `cannot find note with id ${id}` });
+  } else {
+    res.status(200).send(notes[id]);
+  }
+};
+
+const postFunction = async (req, res) => {
+  const fileData = fs.readFileSync('data.json', 'utf8');
+  const data = JSON.parse(fileData);
+  const notes = data.notes;
+  const body = req.body;
+  const writeFileDataJSON = () => fsPromises.writeFile('data.json', JSON.stringify(data, null, 2));
+  try {
+    if (!body.content) {
+      res.status(400).send({ error: 'content is a required field' });
+      return;
+    } else if (body.content) {
+      notes[data.nextId] = body;
+      notes[data.nextId].id = data.nextId;
+      data.nextId++;
+      await writeFileDataJSON();
+      res.status(201).send(body);
+    }
+  } catch (err) {
+    console.error(err);
+    res.status(500).send({ error: 'An unexpected error occured.' });
+  }
+};
+
+const deleteFunction = async (req, res) => {
+  const fileData = fs.readFileSync('data.json', 'utf8');
+  const data = JSON.parse(fileData);
+  const notes = data.notes;
+  const id = Number(req.params.id);
+  const writeFileDataJSON = () => fsPromises.writeFile('data.json', JSON.stringify(data, null, 2));
+  try {
+    if (id < 1 || !Number.isInteger(id) || isNaN(id)) {
+      res.status(400).send({ error: 'id must be a positive integer' });
+    } else if (!notes[id]) {
+      res.status(404).send({ error: `cannot find note with id ${id}` });
+    } else {
+      delete notes[id];
+      await writeFileDataJSON();
+      res.sendStatus(204);
+    }
+  } catch (err) {
+    console.error(err);
+    res.status(500).send({ error: 'An unexpected error occured.' });
+  }
+};
+
+const putFunction = async (req, res) => {
+  const fileData = fs.readFileSync('data.json', 'utf8');
+  const data = JSON.parse(fileData);
+  const notes = data.notes;
+  const id = Number(req.params.id);
+  const body = req.body;
+  const writeFileDataJSON = () => fsPromises.writeFile('data.json', JSON.stringify(data, null, 2));
+  try {
+    if (!body.content) {
+      res.status(400).send({ error: 'content is a required field' });
+      return;
+    }
+    if (id < 1 || !Number.isInteger(id) || isNaN(id)) {
+      res.status(400).send({ error: 'id must be a poitive integer' });
+      return;
+    }
+    if (!notes[id]) {
+      res.status(404).send({ error: `cannot find note with id ${id}` });
+      return;
+    } else if (notes[id]) {
+      notes[id].content = body.content;
+      await writeFileDataJSON();
+      res.status(200).send(notes[id]);
+    }
+  } catch (err) {
+    console.error(err);
+    res.status(500).send({ error: 'An unexpected error occured.' });
+  }
+};
+
+const app = express();
+
+app.use(express.json());
+
+app.get('/api/notes', getFunction);
```

Comment:
This is fine, since it puts your endpoints all together where they're easy to see. But it's also very, very common to just put the endpoints inline as arrow functions.